Makefile.include

```diff
@@ -95,6 +95,20 @@ __DIRECTORY_VARIABLES += \
 
 BOARDSDIR               := $(abspath $(BOARDSDIR))
 
+# Include Board and CPU configuration, if BOARD is not found in BOARDSDIR
+# e.g. when set by the environment fallback to searching in RIOTBOARD
+ifeq (,$(wildcard $(BOARDSDIR)/$(BOARD)))
```

Comment:
For the wildcard,this does not check for a directory, see what was done with `ALLBOARDS` should be `$(wildcard $(BOARDSDIR)/$(BOARD)/.)` I think.
However it's better to use wildcard than the old `$(shell test -d)`.

This was moved before the `global goals` separate handling, so the handling does not seem correct for global targets.
A test with an external board (different name) should show this.

Reply:
By this do you mean that it could ignore the external-boards? This is true, if calling for example `info-board` with and an external `BOARDSDIR` and `BOARDS` set to an internal board, then all `BOARDS` in the external `BOARDSDIR` would be ignored, is this what you meant?

Reply:
Yes as it is done too early, it would overwrite `BOARDSDIR` for the multiple boards case. And so make all the handling done in `makefiles/info-global.inc.mk` and `boards.inc.mk` useless.

Reply:
Addressed and fixed.